src/shared/types/reflect.ts

```diff
@@ -0,0 +1,17 @@
+export type ReflectCursor = {
+  x: number
+  y: number
+} | null
```

Comment:
the `| null` I think should be where `ReflectCursor` type is used so `ReflectCursor | null` and if something is of `ReflectCursor` type then we should expect it to be there